### PR TITLE
Updated CONTRIBUTING.md and MAINTAINERS.md

### PR DESCRIPTION
**Pull request by TheMC47**
### Description

* Fixed broken GitHub links
* Added instructions on how to run tests
* Linked XMonad.Doc.Developing

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've confirmed these changes don't belong in xmonad-contrib instead

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [ ] I updated the `CHANGES.md` file

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -131,11 +145,13 @@ each pull request contains just one commit.
...
+     ```shell
+     $ git push --force-with-lease
+     ```
 
 [xmonad]: https://github.com/xmonad/xmonad
 [xmonad-contrib]: https://github.com/xmonad/xmonad-contrib
 [xmonad-testing]: https://github.com/xmonad/xmonad-testing
 [x11]: https://github.com/xmonad/X11
 [ml]: https://mail.haskell.org/cgi-bin/mailman/listinfo/xmonad
+[xmonad-doc-developing]: http://hackage.haskell.org/package/xmonad-contrib-0.16/docs/XMonad-Doc-Developing.html
```
Comment (liskin) [inline CONTRIBUTING.md]: Make this `http://hackage.haskell.org/package/xmonad-contrib/docs/XMonad-Doc-Developing.html`, hackage will autoredirect to the latest version.

**Review by liskin — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (TheMC47) [reply CONTRIBUTING.md]: Thanks for the tip!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,6 +63,10 @@ Here are some tips for getting your changes merged into xmonad:
     repository.  Include a new configuration file that shows off your
     changes if possible by creating a PR on that repository as well.
 
+  * Make sure you run the automated tests.  Both [xmonad-contrib][]
```
Comment (psibi) [inline CONTRIBUTING.md:66]: I guess you are missing the link ?

Comment (TheMC47) [reply CONTRIBUTING.md:66]: I thought the same, but everything is referenced this way (at the end of the file, all links are defined)

Comment (psibi) [reply CONTRIBUTING.md:66]: TIL, thanks!

**Pull request merged**